tests/test_traintest.py

```diff
@@ -0,0 +1,41 @@
+"""Tests for the s2s.traintest module.
+"""
+import numpy as np
+import pandas as pd
+import pytest
+import s2s.traintest
+from s2s.time import AdventCalendar
+
+class TestTrainTest:
+    # Define all required inputs as fixtures:
+    @pytest.fixture(autouse=True)
+    def dummy_intervals_df(self):
+        freq = "180d"
+        n_intervals = pd.Timedelta("365days") // pd.to_timedelta(freq)
+        # create calendar object
+        cal = AdventCalendar(anchor_date=(10, 15), freq=freq)
+        intervals = cal.map_years(2019, 2021, flat=True)
+        # infer anchor years from give intervals
+        anchor_years = [timestamp.right.year for timestamp in intervals[::n_intervals]]
+        # Create DataFrame from intervals Series to store the train/test groups
+        traintest_base = pd.DataFrame(data = {
+        'anchor_year': np.repeat(anchor_years, n_intervals),
+        'i_intervals': np.tile(range(n_intervals), len(anchor_years)),
+        'intervals': intervals,
```

Comment:
```suggestion
    def dummy_intervals_df(self):
        freq = "180d"
        n_intervals = pd.Timedelta("365days") // pd.to_timedelta(freq)
        # create calendar object
        cal = AdventCalendar(anchor_date=(10, 15), freq=freq)
        intervals = cal.map_years(2019, 2021, flat=True)
        # infer anchor years from give intervals
        anchor_years = [timestamp.right.year for timestamp in intervals[::n_intervals]]
        # Create DataFrame from intervals Series to store the train/test groups
        traintest_base = pd.DataFrame(data = {
        'anchor_year': np.repeat(anchor_years, n_intervals),
        'i_intervals': np.tile(range(n_intervals), len(anchor_years)),
        'intervals': intervals,
```

This is still very specific to the calendar use case, whereas I think in general this function can be much more generic. Could we just generate a random dataframe instead?

Reply:
N/A with the latest updates.